### ISSUE TITLE
settimed archlinux - gives: fatal error: unexpected signal during runtime execution

### ISSUE THREAD

**Issue by Torxed** [xyproto/wallutils#14]
Arch Linux (latest as of today)
WM: awesome

`settimed archlinux` as proposed by https://www.archlinux.org/packages/community/any/archlinux-wallpaper/ does not work.

```
Setting timed wallpaper: archlinux
Searching for wallpapers...
Filtering wallpapers by name...
Using: /usr/share/backgrounds/archlinux/archlinux.stw
Using the Simple Timed Wallpaper format.
Initial static wallpaper event at 12:00
Window: 1h35m37s
Cooldown: 1h35m37s
Filename: /usr/share/backgrounds/archlinux/archlinux-berry.jpg
Setting /usr/share/backgrounds/archlinux/archlinux-berry.jpg.
Using the X11 backend.
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x7f663ec17dca]

runtime stack:
runtime.throw(0x7f663f06994e, 0x2a)
        /usr/lib/go/src/runtime/panic.go:774 +0x74
runtime.sigpanic()
        /usr/lib/go/src/runtime/signal_unix.go:378 +0x480

goroutine 1 [syscall]:
runtime.cgocall(0x7f663f05dbd0, 0xc0002d91e8, 0x1e)
        /usr/lib/go/src/runtime/cgocall.go:128 +0x5d fp=0xc0002d91b8 sp=0xc0002d9180 pc=0x7f663eeb245d
github.com/xyproto/wallutils._Cfunc_SetBackground(0x7f661c000c90, 0x0)
        /tmp/go-build810405401/b067/_cgo_gotypes.go:72 +0x4b fp=0xc0002d91e8 sp=0xc0002d91b8 pc=0x7f663f05946b
github.com/xyproto/wallutils.(*X11).SetWallpaper(0x7f663f2afde0, 0xc00014e6c0, 0x34, 0x0, 0x0)
        /build/wallutils/src/wallutils/x11.go:109 +0x33a fp=0xc0002d9300 sp=0xc0002d91e8 pc=0x7f663f059b0a
github.com/xyproto/wallutils.SetWallpaperCustom(0xc00014e6c0, 0x34, 0x1, 0x7f663f06021d, 0x7, 0xc00026c5b0, 0x0)
        /build/wallutils/src/wallutils/wallpaper.go:45 +0x251 fp=0xc0002d93c8 sp=0xc0002d9300 pc=0x7f663f056dc1
github.com/xyproto/wallutils.SetWallpaperVerbose(...)
        /build/wallutils/src/wallutils/wallpaper.go:77
main.SetTimedWallpaper.func3(0xc00014e6c0, 0x34, 0x7f663f0610f3, 0xc)
        main.go:94 +0x54 fp=0xc0002d9410 sp=0xc0002d93c8 pc=0x7f663f05cac4
github.com/xyproto/simpletimed.(*Wallpaper).SetInitialWallpaper(0xc000200080, 0x7f663f15b701, 0xc000242ad0, 0x7f663f06254f, 0x12, 0xc000242ad0, 0x7f663f06254f)
        /build/wallutils/src/wallutils/vendor/github.com/xyproto/simpletimed/eventloop.go:157 +0x142c fp=0xc0002d97a0 sp=0xc0002d9410 pc=0x7f663f009d2c
github.com/xyproto/simpletimed.(*Wallpaper).EventLoop(0xc000200080, 0xc000242a01, 0xc000242ad0, 0x7f663f06254f, 0x12, 0x1, 0x1)
        /build/wallutils/src/wallutils/vendor/github.com/xyproto/simpletimed/eventloop.go:284 +0x193 fp=0xc0002d9988 sp=0xc0002d97a0 pc=0x7f663f00a493
main.SetTimedWallpaper(0x7ffe9c809d30, 0x9, 0x1, 0x7f663f06254f, 0x12, 0xc0000e5b68, 0x7f663efce20c)
        main.go:94 +0x237 fp=0xc0002d9aa0 sp=0xc0002d9988 pc=0x7f663f05b9d7
main.setTimedWallpaperAction(0xc0000ae160, 0x0, 0xc000096150)
        main.go:123 +0xc1 fp=0xc0002d9b50 sp=0xc0002d9aa0 pc=0x7f663f05c3b1
github.com/urfave/cli.HandleAction(0x7f663f1150a0, 0x7f663f15b8d8, 0xc0000ae160, 0x0, 0x0)
        /build/wallutils/src/wallutils/vendor/github.com/urfave/cli/app.go:514 +0xc0 fp=0xc0002d9b78 sp=0xc0002d9b50 pc=0x7f663efc9050
github.com/urfave/cli.(*App).Run(0xc0000de000, 0xc000096020, 0x2, 0x2, 0x0, 0x0)
        /build/wallutils/src/wallutils/vendor/github.com/urfave/cli/app.go:274 +0x5f8 fp=0xc0002d9ec0 sp=0xc0002d9b78 pc=0x7f663efc6f78
main.main()
        main.go:159 +0x26e fp=0xc0002d9f60 sp=0xc0002d9ec0 pc=0x7f663f05c86e
runtime.main()
        /usr/lib/go/src/runtime/proc.go:203 +0x226 fp=0xc0002d9fe0 sp=0xc0002d9f60 pc=0x7f663eeddfc6
runtime.goexit()
        /usr/lib/go/src/runtime/asm_amd64.s:1357 +0x1 fp=0xc0002d9fe8 sp=0xc0002d9fe0 pc=0x7f663ef092a1

goroutine 19 [syscall]:
os/signal.signal_recv(0x0)
        /usr/lib/go/src/runtime/sigqueue.go:147 +0x9e
os/signal.loop()
        /usr/lib/go/src/os/signal/signal_unix.go:23 +0x24
created by os/signal.init.0
        /usr/lib/go/src/os/signal/signal_unix.go:29 +0x43

goroutine 107 [chan receive]:
github.com/xyproto/simpletimed.(*Wallpaper).EventLoop.func1(0xc00023d260, 0xc000200080, 0x1, 0xc000242ad0, 0x7f663f06254f, 0x12)
        /build/wallutils/src/wallutils/vendor/github.com/xyproto/simpletimed/eventloop.go:269 +0x10a
created by github.com/xyproto/simpletimed.(*Wallpaper).EventLoop
        /build/wallutils/src/wallutils/vendor/github.com/xyproto/simpletimed/eventloop.go:266 +0x13f

```

**Comment by xyproto** [comment]
Thanks for reporting! I'm planning to rewrite `settimed` and fix this.

**Comment by xyproto** [comment]
This should be fixed now, ref #13. Please re-open if not. Thanks.
